sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/BoundedWindow.java

```diff
@@ -21,11 +21,8 @@
 import org.joda.time.Instant;
 
 /**
- * A {@code BoundedWindow} represents a finite grouping of elements, with an
- * upper bound (larger timestamps represent more recent data) on the timestamps
- * of elements that can be placed in the window. This finiteness means that for
- * every window, at some point in time, all data for that window will have
- * arrived and can be processed together.
+ * A {@code BoundedWindow} represents window information assigned to data elements.
+ * At some point in time, all data with the same window will arrive and can be processed together.
```

Comment:
I think if you remove the discussion of "upper bound" I think replace it with something, since having a `maxTimestamp()` is the main feature of windows.

Basically the `maxTimestamp()` is the default event time where the window is expected to be complete, and it can be garbage collected unless allowed lateness is set.